File: browser/src/main/java/com/sys1yagi/swipe/browser/view/SwipeViewPagerCountAdapter.java
package com.sys1yagi.swipe.browser.view;

import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;

public class SwipeViewPagerCountAdapter extends PagerAdapter {

    int count;

    public SwipeViewPagerCountAdapter(int count) {
        this.count = count;
    }

    @Override
    public int getCount() {
        return count;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        // no op
        View view = new View(container.getContext());
        container.addView(view);
        return null;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        // no op
        container.removeView((View) object);
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }
}

File: core/src/main/java/com/sys1yagi/swipe/core/view/SwipeViewPagerCountAdapter.kt
package com.sys1yagi.swipe.core.view

import android.support.v4.view.PagerAdapter
import android.view.View
import android.view.ViewGroup

class SwipeViewPagerCountAdapter(internal var count: Int) : PagerAdapter() {

    override fun getCount(): Int {
        return count
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any? {
        // no op
        val view = View(container.context)
        container.addView(view)
        return null
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any?) {
        // no op
        `object`?.let {
            container.removeView(it as View)
        }
    }

    override fun isViewFromObject(view: View, `object`: Any?): Boolean {
        return view === `object`
    }
}
